src/components/CountryDetails.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router-dom'
+
+function CountryDetails(props) {
+
+    const { countryId } = useParams()
+
+    const [currentCountry, setCurrentCountry] = useState(null)
+
+    useEffect(() => {
```

Comment:
no needs to useEffect and state here, you can take props directly and use it,

     let currentCountry= props.countries.find((e) => { return e.alpha3Code == countryId })